include/mrdocs/Config.hpp

```diff
@@ -104,6 +104,10 @@ class MRDOCS_DECL
         */
         std::string extraYaml;
 
+        /** A string holding the complete CMake arguments.
+        */
+        std::string cmake;        
```

Comment:
Everything should go into `ConfigImpl` unless external plugins need it (of we don't have evidence yet). 

But in that case you need neither `Config` or `ConfigImpl`. You can get this value directly from `Config::object`.